### DISCUSSION TITLE
Node coordinates appear blocky

### DISCUSSION THREAD

**Discussion by ndicolaNIH** [talmolab/sleap#2006] [Help!]
Hello, I apologize if this has been covered already but I couldn't find a help discussion for it. I am running SLEAP on two rats who can have overlapping frames. I am analyzing the data from predicted frames and I've noticed that the coordinates appear almost discretized, or "blocky". The amount of "blockyness" is different for the anchor point and the other nodes. Below is a map of coordinates for the shoulders (the anchor point) and the nose.
![sleap-tracking-jump_shoulders](https://github.com/user-attachments/assets/1f038a91-2482-49dd-8892-a1e6a5e0a2d8)
![sleap-tracking-jump_nose](https://github.com/user-attachments/assets/76f5fc17-bcda-4b9d-8919-843256da50a0)
A few things I've seen suggested for other problems are that the tail base is a better anchor point because it is closer to the "center" of the skeleton, but the tail base is visible less than the shoulders. I've also seen another discussion where increasing the sigma values can lead to blockyness, but I have not changed my sigma values from the default 2.5 for either the centroid model or centered instance model. Is this a problem that can be solved by turning down that value to 2.1 or 2.2? I've seen a comment not to lower it <2. Additionally, in the centroid model page you'll see that the receptive field for the centroid doesn't perfectly align to the corners of the environment since the camera is tilted. Is that okay? I couldn't tell if that was going to cause problems detecting the centroid in the bottom left corner.
Here are screenshots of my training setup and I've included the training data for both models as well.
![example-centroid-model](https://github.com/user-attachments/assets/0dadd08c-a8be-4849-87c3-5d53bedba0ba)
![example-sleap](https://github.com/user-attachments/assets/83dbd172-6ec4-47b3-83f0-5b4127bdf183)
![example-training-pipeline](https://github.com/user-attachments/assets/a7dea12c-35ac-4e01-a846-4e4ef2b5818b)
[training_log_centroid.csv](https://github.com/user-attachments/files/17510488/training_log_centroid.csv)
[training_log_instance.csv](https://github.com/user-attachments/files/17510489/training_log_instance.csv)

**Comment by eberrigan** [comment]
Hi @ndicolaNIH,

The receptive field size should be around the size of your animal for the centroid model. That will be used to crop around your animal and provide that to the centered instance model. Please increase your input scaling and let us know if that fixes your issue.

https://sleap.ai/guides/choosing-models.html#choosing-models

Best,

Elizabeth

**Comment by talmo** [comment]
Hi @ndicolaNIH,

This is an artifact known as quantization error, which happens due to the limitations in our ability to resolve coordinates below a certain resolution. See this discussion comment for more details and let us know if you have any questions: https://github.com/talmolab/sleap/discussions/1739#discussioncomment-9064817

Cheers,

Talmo
